lib/tom_queue/deferred_work_manager.rb

```diff
@@ -80,13 +80,19 @@ def schedule(response, headers, payload)
     def start
       debug "[DeferredWorkManager] Deferred process starting up"
 
+      register_signal_handlers
```

Comment:
No. Just register the signal handlers in the `daemon` class, and call the `stop` method from there.